extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get Centroid.pushbutton/script.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import print_function
+from pyrevit import script, revit, DB
+from Autodesk.Revit.Exceptions import InvalidOperationException
+
+doc = revit.doc
+uidoc = revit.uidoc
+
+logger = script.get_logger()
+output = script.get_output()
+
+
+def merge_solids(solids):
+    """Merges the provided list of solids into a single unified solid
+
+    Args:
+        solids (list(DB.Solid)): The solids to merge
+
+    Raises:
+        ValueError: If the input is an empty list
+
+    Returns:
+        DB.Solid: The union of the solids
+    """
+    union = None
+    if not solids:
+        raise ValueError('List of provided solids was empty.')
+    for solid in solids:
+        if not union:
+            union = solid
+        else:
+            try:
+                union = DB.BooleanOperationsUtils.ExecuteBooleanOperation(
+                    union,
+                    solid,
+                    DB.BooleanOperationsType.Union
+                )
+            except InvalidOperationException:
+                ref = solid.Faces[0].Reference.ConvertToStableRepresentation(doc)
+                uid = ref.split(':')[0]
+                e = doc.GetElement(uid)
+                logger.error(
+                    'Failed to merge solid from element {}, '
+                    'it will be excluded from the calculation'.format(
+                        revit.ElementWrapper(e).name
+                    )
+                )
+    return union
+
+
+selection = revit.get_selection()
+if not selection:
+    forms.alert('You must select one element.', exitscript=True)
```

Comment:
added the error handling when nothing is selected

Reply:
or we could just add the selection context to the bundle. would you?